### PR TITLE
[Merged by Bors] - feat(Algebra.Order.Star.Basic): add `aesop` lemma  `IsSelfAdjoint.mul_self_nonneg`.

### PR DESCRIPTION
**Pull request by JonBannon**
Includes the simple fact that in a `StarOrderedRing`, selfadjoint elements have nonnegative squares. 

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Order/Star/Basic.lean [hunk 1]
```diff
@@ -166,6 +166,10 @@ theorem star_mul_self_nonneg (r : R) : 0 ≤ star r * r :=
 theorem mul_star_self_nonneg (r : R) : 0 ≤ r * star r := by
   simpa only [star_star] using star_mul_self_nonneg (star r)
 
+@[aesop safe apply (rule_sets := [CStarAlgebra])]
+theorem IsSelfAdjoint.mul_self_nonneg {a : R} (ha : IsSelfAdjoint a) : 0 ≤ a * a := by
+  simpa [isSelfAdjoint_iff |>.mp ha] using star_mul_self_nonneg a
```
Comment (j-loreaux) [inline Mathlib/Algebra/Order/Star/Basic.lean:171]:
```suggestion
  simpa [ha.star_eq] using star_mul_self_nonneg a
```

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

File: Mathlib/Algebra/Order/Star/Basic.lean [hunk 2]
```diff
@@ -166,6 +166,10 @@ theorem star_mul_self_nonneg (r : R) : 0 ≤ star r * r :=
 theorem mul_star_self_nonneg (r : R) : 0 ≤ r * star r := by
   simpa only [star_star] using star_mul_self_nonneg (star r)
 
+@[aesop safe apply (rule_sets := [CStarAlgebra])]
+theorem IsSelfAdjoint.mul_self_nonneg {a : R} (ha : IsSelfAdjoint a) : 0 ≤ a * a := by
+  simpa [ha.star_eq] using star_mul_self_nonneg a
```
Comment (j-loreaux) [inline Mathlib/Algebra/Order/Star/Basic.lean:171]: Thanks! Can you add a `IsSelfAdjoint.sq_nonneg : 0 ≤ a ^ 2` version too? In that case you'll need to assume the `R` is unital (you can put the result in the `Semiring` section below.

**Review by j-loreaux — COMMENTED** [review]
bors d+

File: Mathlib/Algebra/Order/Star/Basic.lean (hunk 2 above)
Comment (JonBannon) [reply Mathlib/Algebra/Order/Star/Basic.lean:171]: Will do this before finishing the PR. Thanks!

**Pull request closed without merging**